File: generate.web/ClientApp/src/app/models/app/categorySetDto.ts
import { CategoryOptionDto } from './categoryOptionDto';
import { TableType } from './tableType';

export interface CategorySetDto {
    categorySetId: number;
    organizationLevelCode: string;
    submissionYear: string;
    categorySetCode: string;
    categorySetName: string;
    viewDefinition: string;
    includeOnFilter: string;
    excludeOnFilter: string;
    categories: Array<string>;
    categoryOptions: Array<CategoryOptionDto>;
    tableTypes: Array<TableType>;
}
export interface Categoriess {
    categoryId: number;
    categoryName: string;

}

export interface Filter {
    filterCode: number;
    filterName: string;

}

export interface CatToDisplay {
    categorySetCode: number;
    categorySetName: string;

}

File: generate.core/Dtos/App/CategorySetDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace generate.core.Dtos.App
{
    public class CategorySetDto
    {
        public int CategorySetId { get; set; }
        public string OrganizationLevelCode { get; set; }
        public string SubmissionYear { get; set; }
        public string CategorySetCode { get; set; }
        public string CategorySetName { get; set; }
        public string ViewDefinition { get; set; }
        public string ExcludeOnFilter { get; set; }
        public string IncludeOnFilter { get; set; }
        public List<string> Categories { get; set; }
        public List<CategorySetCategoryOptionDto> CategoryOptions { get; set; }
    }
}
